src/types/auth.d.ts

```diff
@@ -0,0 +1,23 @@
+import { UserCredential } from 'firebase/auth';
+
+export declare namespace Auth {
+  export interface IUser {
+    uid: string;
+    email: string;
+  }
+  export interface ILoginData {
+    email: string;
+    password: string;
+  }
+  export interface ISignUpData extends ILoginData {
+    repeatPassword: string;
+  }
```

Comment:
I'd refactor this slightly to get rid of some duplication in the form component:
```suggestion
  export interface IFormData {
    email: string;
    password: string;
    repeatPassword?: string;
  }
```